tests/test_rag_tools.py

```
@@ -0,0 +1,290 @@
+        # Test invalid limit
+        with pytest.raises(Exception):
+            await mcp_client.call_tool(
+                "rag_query",
+                {
+                    "query": "test",
+                    "limit": 0,
+                },
+            )
+
+        # Test negative offset
+        with pytest.raises(Exception):
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)